

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Tripal 3 Roadmap &mdash; KnowPulse  documentation</title>
  

  
  
  
  

  

  
  
    

  

  <link rel="stylesheet" href="../_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="../_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="../_static/theme_overrides.css" type="text/css" />
    <link rel="index" title="Index" href="../genindex.html" />
    <link rel="search" title="Search" href="../search.html" /> 

  
  <script src="../_static/js/modernizr.min.js"></script>

</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">

    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search">
          

          
            <a href="../index.html" class="icon icon-home"> KnowPulse
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="../search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">Contents:</span></p>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../developer_docs.html">KnowPulse Developer Documentation</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="../index.html">KnowPulse</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="../index.html">Docs</a> &raquo;</li>
        
      <li>Tripal 3 Roadmap</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="../_sources/developer_docs/tripal3_roadmap.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="tripal-3-roadmap">
<h1>Tripal 3 Roadmap<a class="headerlink" href="#tripal-3-roadmap" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li>The upgrade from Tripal2 to Tripal 3 requires a complete re-write of all our content pages. This is necessitated by the switch from templates to fields.</li>
<li>We are using this upgrade process to redesign, refresh and rethink the current content pages on KnowPulse to ensure they are the most helpful they can be</li>
<li>This document will show a breakdown of all planned Tripal Content Types for KnowPulse in Tripal3, as well as, include detailed plans for field development in each type.</li>
<li>The following plans will not all be implemented in the first release of KnowPulse on Tripal3 but instead represent the complete picture of our current plans</li>
<li>Please note that the move from Tripal2 to Tripal3 represents a switch from chado-based content types to ontology-based content types. As such one content type in Tripal2 will become multiple specialized content types in Tripal3.</li>
</ul>
</div>
<div class="section" id="knowpulse-content-types">
<h1>KnowPulse Content Types<a class="headerlink" href="#knowpulse-content-types" title="Permalink to this headline">¶</a></h1>
<ul class="simple">
<li>chado.organism table (seperated using organismprop)
- Crop
- Wild Species
- Related Species</li>
<li>chado.project table (seperated using projectprop)
- Research Project (semantic science ontology; study; SIO:001066)
- Research Experiment (semantic science ontology; experiment; SIO:000994)
- Genome Assembly (Sequence Ontology; genome)</li>
<li>chado.publication (full table)
- Publication (TPUB; Publication)</li>
<li>chado.feature table (seperated using feature.type_id)
- Chromosome (Sequence Ontology; pseudomolecule)
- Contig (Sequence Ontology; contig)
- Variant (Sequence Ontology; sequence_variant)
- Genetic Marker  (Sequence Ontology; genetic_marker)
- QTL (Sequence Ontology; QTL)</li>
<li>chado.stock (seperated using stock.type_id)
- Germplasm Accession (Crop ontology; accession; CO_010:0000044)
- Breeding Cross (Crop ontology; generated germplasm(CO_010:0000255)?; breeder’s line(CO_010:0000093)?, breeding material(CO_010:0000159)?)
- Registered Varieties (Sequence Ontology; cultivar; CO_010:0000029)
- Recombinant Imbred Line (Crop Ontology; inbred line; CO_010:0000162)</li>
<li>chado.cvterm (specific cv_id)
- Phenotypic Trait (Trait Ontology; plant trait)</li>
<li>chado.featuremap (full table)
- Genetic Map (data onotology; Map)</li>
</ul>
<p>ADD STOCK COLLECTIONS (i.e. diversity panels)</p>
<div class="section" id="general">
<h2>General<a class="headerlink" href="#general" title="Permalink to this headline">¶</a></h2>
<div class="section" id="species-organism">
<h3>Species (Organism)<a class="headerlink" href="#species-organism" title="Permalink to this headline">¶</a></h3>
<p>The organism content type will include each of our crops and wild relatives. Stored in the Chado Organism table. Design-wise we are thinking of these as launch pads which both summarize the data per crop and link to pre-filtered searches.</p>
<div class="section" id="genome-pane">
<h4>Genome Pane<a class="headerlink" href="#genome-pane" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li>Current Assembly: This field should provide information on the most recent genome assembly or assemblies for this species.</li>
<li>Genome Browser: This field should summarize the data in the current JBrowse and have an image linking to the JBrowse instance for the current assembly</li>
<li>Whole-Genome Diagram: This field should show an embedded CViTjs diagram summarizing the current assembly’s annotation and markers.</li>
<li>Past Assemblies: This field should list the past assemblies with links to archived information.</li>
</ul>
</div>
<div class="section" id="genetic-markers-pane">
<h4>Genetic Markers Pane<a class="headerlink" href="#genetic-markers-pane" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li>Types: This field summarizes the types of markers available for this organism.</li>
<li>Quick Search: This field provides a simple keyword search with limited advanced search collapsed. When the user clicks search, they are redirected to the marker search page with their results pre-filtered.</li>
</ul>
</div>
<div class="section" id="genetic-maps-qtl-pane">
<h4>Genetic Maps &amp; QTL Pane<a class="headerlink" href="#genetic-maps-qtl-pane" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li>Tiled display of available maps with each tile containing map metadata such as population, date generated, number of markers, QTL list</li>
<li>QTL Quick Search: simple keyword search for QTLs.</li>
</ul>
</div>
<div class="section" id="germplasm-pane">
<h4>Germplasm Pane<a class="headerlink" href="#germplasm-pane" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li>Types: This field summarizes the types of germplasm available for this organism. Should this be subject to permissions?</li>
<li>Quick Search: This field provides a simple keyword search. No need for advanced search or breakdown by type.</li>
<li>Recent UofS Varieties: a list of the most recent varieties.</li>
<li>Variety Spotlight: Highlight a single variety</li>
<li>RIL Populations: This could be a cool search that lets them select things like interspecific, species, phenotype of interest. Essentially allow people to search for a RIL without knowing it’s number</li>
<li>Germplasm Diversity: Show a map with each of our germplasm indicated.</li>
<li>Wild Relatives: summarize the gene pools and/or phylogeny</li>
</ul>
</div>
<div class="section" id="phenotypes">
<h4>Phenotypes<a class="headerlink" href="#phenotypes" title="Permalink to this headline">¶</a></h4>
<ul class="simple">
<li>Quick Search: This field provides a simple keyword search. No need for advanced search? It would be helpful this field searched both names and linked cvterms.</li>
<li>Phenotype Projects: Highlight our biggest and/or most recent phenotyping projects. Should show metadata for each such as population phenotyped, purpose, siteyears w/ data, funders logos if applicable. Links to project pages.</li>
</ul>
</div>
</div>
<div class="section" id="research-project">
<h3>Research Project<a class="headerlink" href="#research-project" title="Permalink to this headline">¶</a></h3>
<p>Currently just a regular old project page. Stored in the Chado Project table. Need a way to break out? What about datasets which are currently projects with data? These pages should include all metadata about the project and summarize all the data generated by it.</p>
<p>### Overview Pane
- Project meta-data (stored as project properties)
- Abstract
### Genomic Data
- Barchart or infographic summarizing the types and amount of genomic data (sequence, genes, etc.) generated as part of this project.
### Genotypic Data
- Barchart or infographic summarizing the types and amount of genotypic data (markers, variants, germplasm assayed, genotype calls, etc.) for this project.
- Genotypes Search: Redirects to the genotype matrix pre-filtered for the current project. Provides form elements to further restrict the matrix.
### Phenotype Data
- Barchart or infographic summarizing the siteyears, traits and germplasm phenotyped for this project.
### Germplasm
- List the germplasm associated with this project with download. It might be a long list so make it searchable. Also, we should support stock collections which would then be listed at the top. For example, AGILE is using the Lentil Diversity Panel so it would be nice to say that at the top so users know it’s not a random list ;-).
- We also need a way to add germplasm to a project through the form so keep in mind this one will need a form element.</p>
</div>
<div class="section" id="publication">
<h3>Publication<a class="headerlink" href="#publication" title="Permalink to this headline">¶</a></h3>
<p>Stored in the Chado Publication table.</p>
<p>### Genome Assembly
Not sure where to store this information yet… Perhaps a type of project but the project table doesn’t have a type_id. Perhaps an analysis but then there will be a bunch of useless fields?</p>
</div>
</div>
<div class="section" id="genomic">
<h2>Genomic<a class="headerlink" href="#genomic" title="Permalink to this headline">¶</a></h2>
<p>The following Tripal Content types refer to Genomic Content. All of the following content types are stored in the Chado Feature table. The specific records per type are indicated via the feature.type_id column.</p>
<div class="section" id="chromosome-contig">
<h3>Chromosome/Contig<a class="headerlink" href="#chromosome-contig" title="Permalink to this headline">¶</a></h3>
<p>The following described any page for a backbone sequence. Stored in the Chado Feature table broken out by type_id. These are currently unhelpful feature pages and new functionality to make them more useful is needed. That said, these pages likely won’t get used directly very often.</p>
</div>
<div class="section" id="variant">
<h3>Variant<a class="headerlink" href="#variant" title="Permalink to this headline">¶</a></h3>
<p>The variant page is currently a customized feature page. Stored in the Chado Feature table broken out by type_id.</p>
</div>
<div class="section" id="genetic-marker">
<h3>Genetic Marker<a class="headerlink" href="#genetic-marker" title="Permalink to this headline">¶</a></h3>
<p>The genetic marker page is currently a customized feature page. Stored in the Chado Feature table broken out by type_id.</p>
</div>
</div>
<div class="section" id="germplasm">
<h2>Germplasm<a class="headerlink" href="#germplasm" title="Permalink to this headline">¶</a></h2>
<p>The following Tripal Content types refer to Germplasm Content. All of the following content types are stored in the Chado Stock table. The specific records per type are indicated via the stock.type_id column.</p>
<div class="section" id="germplasm-accession">
<h3>Germplasm Accession<a class="headerlink" href="#germplasm-accession" title="Permalink to this headline">¶</a></h3>
<p>These are currently customized stock pages managed by the Tripal Germplasm module. Stored in the Chado Stock table broken out by type_id.</p>
<p>### Overview
- Basic information such as name, accession, what-not. This will remain a default Tripal pane
- It would be great to finally have the breeder short-hand for the pedigree! This could be a separate field from the pedigree diagram so I can put it in the overview.</p>
<p>### Pedigree
- Shows the parentage pedigree diagram with all but the first 3 levels collapsed.
- All diagrams should now include a figure legend.
### Genotypes Pane
- Types: This field summarizes the number of markers per type with calls for the current germplasm.
- Genotypes Search: Redirects to the genotype matrix pre-filtered for the current germplasm. Provides form elements to further restrict the matrix.
### Breeding Markers
- List the calls for a set of CORE breeding markers. Make the set configurable at the field level.
### Phenotypes Pane
- List the value for a set of CORE phenotypes (e.g. seed coat, cotyledon color). Make the set configurable at the field level. Should it be entereable on the form? How do we handle multiple experiments, siteyears for a single trait/variety combo.</p>
</div>
<div class="section" id="cultivated-variety">
<h3>Cultivated Variety<a class="headerlink" href="#cultivated-variety" title="Permalink to this headline">¶</a></h3>
<p>These are currently customized stock pages managed by the Tripal Germplasm module. Stored in the Chado Stock table broken out by type_id. See <a class="reference external" href="https://www.crop-diversity.org/mgis/accession/01BEL084245">https://www.crop-diversity.org/mgis/accession/01BEL084245</a>
### SAME AS ACCESSION
- Overview will indicate specific variety information and include the abstract and link to the registration publication if known.
- Phenotype pane should highlight phenotypes specifically of interest for this cultivar.</p>
</div>
<div class="section" id="breeding-cross">
<h3>Breeding Cross<a class="headerlink" href="#breeding-cross" title="Permalink to this headline">¶</a></h3>
<p>These are currently customized stock pages managed by the Tripal Germplasm module. Stored in the Chado Stock table broken out by type_id.</p>
<p>### SAME AS ACCESSION
- Overview will include information about the crossing block
- Is there enough information for a Crossing Block Pane? We would have year, season, number of crosses made…</p>
</div>
<div class="section" id="recombinant-imbred-line">
<h3>Recombinant Imbred Line<a class="headerlink" href="#recombinant-imbred-line" title="Permalink to this headline">¶</a></h3>
<p>These are currently customized stock pages managed by the Tripal Germplasm module. Stored in the Chado Stock table broken out by type_id.
### SAME AS ACCESSION
- This Content Type will include all the panes that the Accession Content Type does.
- Overview should include specifics about the RIL including parents, latest generation, status and number of individuals.
- Phenotype pane should highlight phenotypes specifically of interest for this RIL.
### RIL Development Pane
- Reiterate the type of cross (Interspecific, intraspecific)
- Indicates the parents of the RIL including their species.
- Development timeline including which years &amp; how many individuals
- If the RIL is complete, indicate that and the final number of individuals.
### RIL Genotypes Pane
- Links to the VCF Filter form for the current RIL to facillitate downloading of the data.
- Field Provided by VCF Filter.</p>
</div>
</div>
</div>
<div class="section" id="knowpulse-field-types">
<h1>KnowPulse Field Types<a class="headerlink" href="#knowpulse-field-types" title="Permalink to this headline">¶</a></h1>
<p>The following section described common field types.
## Quick Search (Should be in Core?)
- This field type allows a user to search for additional content quickly and easily.
- It consists of a single keyword search box and optional, configurable advanced search form.
- Configuration:</p>
<blockquote>
<div><ul class="simple">
<li>Search URL this field redirects to</li>
<li>what key to use in the query string for entered keywords</li>
<li>Drupal Form API definition of advanced search elements where the keys of the elements are used in the query string</li>
<li>Description and/or help text to display to the user</li>
<li>Quick Search Title</li>
</ul>
</div></blockquote>
<p>## Summary Field (Should be in Core?)
- This field displays a graphical summary based on a materialized view.
- Configuration:</p>
<blockquote>
<div><ul class="simple">
<li>Select the Materialized View to get the summary from</li>
<li>Select which mview column maps to the legend (e.g. type name)</li>
<li>Optional filter by field: select field and mview column mapping</li>
<li>Select chart type: Pie, bar, etc.</li>
<li>Figure Title</li>
<li>Figure Description</li>
<li>Chart type specific configuration</li>
</ul>
</div></blockquote>
<p>## Embedded JBrowse by organism
- Embeds a JBrowse instance
- Provided by the JBrowse integration module
- Contains linkout to full JBrowse page for more intensive browsing
- Configuration:</p>
<blockquote>
<div><ul class="simple">
<li>Select Organism Field for the current content type</li>
<li>Show/Hide Menubar, search, track selector</li>
<li>For each organism: select jbrowse, tracks, and location to show</li>
</ul>
</div></blockquote>
<p>## Whole-Genome View by organism
- Embeds a CViTjs diagram
- Provided by the CViTjs integration module
- Contains linkout to full diagram
- Configuration:</p>
<blockquote>
<div><ul class="simple">
<li>Select Organism Field for the current content type</li>
<li>Show/Hide Controls, track selector</li>
<li>For each organism: select diagram and tracks to show</li>
</ul>
</div></blockquote>
<p># Miscellaneous Notes
- We need an assembly page. Should this be an analysis or a project?
- We need to consider whether the individuals in a RIL need a page.</p>
</div>


           </div>
           
          </div>
          <footer>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2018, LA Sanderson, CT Caron, R Tan, Y Shen, KE Bett (University of Saskatchewan, Pulse Bioinformatics).

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  

    <script type="text/javascript">
        var DOCUMENTATION_OPTIONS = {
            URL_ROOT:'../',
            VERSION:'',
            LANGUAGE:'None',
            COLLAPSE_INDEX:false,
            FILE_SUFFIX:'.html',
            HAS_SOURCE:  true,
            SOURCELINK_SUFFIX: '.txt'
        };
    </script>
      <script type="text/javascript" src="../_static/jquery.js"></script>
      <script type="text/javascript" src="../_static/underscore.js"></script>
      <script type="text/javascript" src="../_static/doctools.js"></script>
      <script type="text/javascript" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.1/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>

  

  <script type="text/javascript" src="../_static/js/theme.js"></script>

  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>